libtransmission/crypto-utils.cc

```diff
@@ -106,15 +106,15 @@ int tr_rand_int_weak(int upper_bound)
 {
     TR_ASSERT(upper_bound > 0);
 
-    static bool init = false;
-
-    if (!init)
+    static std::mt19937 random_engine = []()
```

Comment:
Not sure why lambda is necessary here.
```suggestion
    static auto random_engine = std::mt19937{ std::random_device{}() };
```